src/Mod/Path/PathScripts/PathDrilling.py

```diff
@@ -116,20 +115,14 @@ def circularHoleExecute(self, obj, holes):
         self.commandlist.append(Path.Command('G90'))
         self.commandlist.append(Path.Command(obj.ReturnLevel))
 
-        # ml: I'm not sure whey these were here, they seem redundant
-        # # rapid to first hole location, with spindle still retracted:
-        # p0 = holes[0]
-        # self.commandlist.append(Path.Command('G0', {'X': p0['x'], 'Y': p0['y'], 'F': self.horizRapid}))
-        # # move tool to clearance plane
-        # self.commandlist.append(Path.Command('G0', {'Z': obj.ClearanceHeight.Value, 'F': self.vertRapid}))
-
         for p in holes:
             cmd = "G81"
             cmdParams = {}
             cmdParams['Z'] = p['trgtDep'] - tiplength
             cmdParams['F'] = self.vertFeed
-            cmdParams['R'] = obj.RetractHeight.Value
+            cmdParams['R'] = obj.SafeHeight.Value
             if obj.PeckEnabled and obj.PeckDepth.Value > 0:
+                cmdParams['R'] = obj.PeckRetractHeight.Value
```

Comment:
Given this change I think it would be great if the UI would disable the fields if pecking is not enabled. When adding that, would it be possible you could also add that behaviour to the dwell time?

Reply:
Done, also dwell and peck are now exclusive in the gui as we can't generate gcode for a peck and dwell cycle.